File: erp-purchase/src/main/java/com/own/erp/purchase/entity/Supplier.java
package com.own.erp.purchase.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * @author : chenyi
 * @Date : 2026/9/3
 * @Description : 供应商(supplier)
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("supplier")
public class Supplier {

    /** 主键 */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 供应商名称 */
    private String name;

    /** 联系人 */
    private String contact;

    /** 联系电话 */
    private String phone;

    /** 结算方式,走 sys_dict(预付/月结等) */
    private String settleType;

    /** 备注 */
    private String remark;

    /** 1启用0禁用 */
    private Integer status;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;

    /** 逻辑删除:0=正常,非0=已删(值=被删行id);delval=id 配合唯一键含 deleted,删后同键可重建(TODO#7) */
    @TableLogic(value = "0", delval = "id")
    private Long deleted;
}

File: erp-web/src/api/interface/purchase/supplier.ts
/**
 * 本文件由 pnpm gen:page 生成(spec 行式拍板 + tools/openapi.json 快照)
 * 默认存在即跳过:人工改动不会被 --force 之外的任何方式覆盖;重新生成前先 diff 人工改动
 * 框架代码禁手改;业务槽位一律 TODO(编号),编号已登记 TODO.md
 */

/** 供应商管理接口类型(gen:page 从 openapi 快照生成;对齐后端契约,禁手抄字段) */

/** 供应商管理实体(分页行 / 详情) */
export interface SupplierResponse {
  /** id */
  id: number
  /** 供应商名称 */
  name: string
  /** 联系人 */
  contact: string
  /** 联系电话 */
  phone: string
  /** 结算方式 */
  settleType: string
  /** 备注 */
  remark: string
  /** 状态 */
  status: number
  /** 创建时间 */
  createdAt: string
  /** updatedAt */
  updatedAt: string
}

/** 供应商管理新增/修改入参 */
export interface SupplierSaveRequest {
  /** name */
  name?: string
  /** contact */
  contact?: string
  /** phone */
  phone?: string
  /** settleType */
  settleType?: string
  /** remark */
  remark?: string
  /** status */
  status?: number
}
